File: c7-8/file.c
#include <fcntl.h>
#include <unistd.h>
#include<stdio.h>

int main() {
    // 创建文件（如果不存在），返回文件描述符
    int fd = open("newfile.txt", O_CREAT | O_WRONLY, 0644);

    // 打开文件
    int fd1 = open("file.txt", O_WRONLY | O_APPEND);

    for line in f:
        print(line)
    // 写入
    write(fd, "Hello", 5);  //覆盖
    write(fd1, "Hello", 5); //追加
    // 关闭
    close(fd);

    // 删除文件用的是unlink而不是delete
    unlink("file_to_delete.txt");  

    // 移动/重命名
    rename("oldname.txt", "newname.txt");

    // 硬链接
    link("original.txt", "hardlink.txt");

    //软链接
    symlink("target.txt", "symlink.txt");

    return 0;
}

File: c7-8/file.py
import os

# 创建空文件
open("newfile.txt", "w").close()

# 创建读写内容
with open("newfile.txt", "w") as f:
    content = f.read()
    f.write("Hello World")
    for line in f:
        print(line)

# 创建并指定属性
dst_fd = os.open("filename.txt", os.O_WRONLY | os.O_CREAT | os.O_TRUNC, 0o644)

# 删除文件
os.remove("file_to_delete.txt")

# 移动/重命名文件
os.rename("oldname.txt", "newname.txt")

# 链接文件（硬链接）
os.link("original.txt", "hardlink.txt")

# 链接文件（软链接）
os.symlink("original.txt", "softlink.txt")